src/server/handlers/join.rs

```diff
@@ -40,15 +40,19 @@ impl JsonRPCRequestHandler for JoinSubnetHandler {
     type Response = ();
 
     async fn handle(&self, request: Self::Request) -> anyhow::Result<Self::Response> {
-        let conn = match self.pool.get(&request.subnet) {
+        let subnet = SubnetID::from_str(&request.subnet)?;
+        let parent = subnet
+            .parent()
+            .ok_or_else(|| anyhow!("no parent found"))?
+            .to_string();
+        let conn = match self.pool.get(&parent) {
```

Comment:
Wouldn't it make more sense for `pool.get` to accept a `SubnetID` as an argument instead of a string? It would remove the need of an additional check inside of get and having to do a lot of back and forth between types.

Reply:
yeah, make sense, haha. Update this now.